### PR TITLE
fix: batch get skills when there are many skill ids

### PR DESCRIPTION
**Pull request by liuliquan**

### CODE REVIEW

File: src/common/helper.js [hunk 1]
```diff
@@ -1139,16 +1139,41 @@ async function getMembersByHandles(handles) {
...
+          {
+            headers: { Authorization: `Bearer ${token}` },
+          }
+        );
+        return res.data;
+      })()
+    );
+  }
+
+  const datas = await Promise.all(promises);
```
Comment (Aranyajit) [inline src/common/helper.js:1175]: data is plural by itself, change to data or skillData

File: src/common/helper.js [hunk 2]
```diff
@@ -1139,16 +1139,41 @@ async function getMembersByHandles(handles) {
  * @returns {Object}
  */
 async function getStandSkills(ids) {
```
Comment (Aranyajit) [inline src/common/helper.js:1141]: Does this have an error handling from where it is called? If not, you need to handle the error to rollback the entire transaction if any of the batches are unsucessful in returning skill data

File: src/common/helper.js [hunk 3]
```diff
@@ -1139,16 +1139,41 @@ async function getMembersByHandles(handles) {
...
+    if (queryString.length + skillIdArg.length + enid.length < 2048) {
+      queryString += skillIdArg + enid;
+    } else {
+      queryBatches.push(queryString);
+      queryString = "disablePagination=true" + skillIdArg + enid;
+    }
+  }
+  queryBatches.push(queryString);
+
+  const promises = [];
```
Comment (Aranyajit) [inline src/common/helper.js:1159]: rename to skillDataPromises please to enhance readability

**Review by Aranyajit — CHANGES_REQUESTED** [review]
Please check my comments

File: src/common/helper.js (hunk 2 above)
Comment (liuliquan) [reply src/common/helper.js:1141]: It's called only by [validateSkills](https://github.com/topcoder-platform/challenge-api/blob/dev/src/common/challenge-helper.js#L107) to validate skill ids before the actual update

**Review by Aranyajit — APPROVED** [review]
Approved

**Review by Aranyajit — APPROVED** [review]
Approved

**Review by eisbilir — APPROVED** [review]

**Pull request merged**